LocalStorage/LocalStorageError.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  RealmDBError.swift
+//  WeatherDemoApp
+//
+//  Created by ahmed on 06/04/2024.
+//
+
+import Foundation
+
+enum LocalStorageError: LocalizedError {
+    case addError
+    case deleteError
+    case observationError
```

Comment:
```suggestion
    case addError(underlyingError: Error)
    case deleteError(underlyingError: Error)
    case observationError(underlyingError: Error)
```

A nice approach is to include the underlying error as well. This gives you more context why this happened.